docsifyjs/docsify
# Chore: Replace live-server, clean up server implementation, and update test docs

## Summary

This PR cleans up our serve configuration as discussed in #2104 and #2218:

- Standardize on a single project server by replacing [live-server](https://www.npmjs.com/package/live-server) with [BrowserSync](https://browsersync.io) for development server
- Consolidate server configurations into single file for easier management (`server.config.js`)
- Update server configurations to allow multiple  servers to run simultaneously:
  - Development: 3000
  - Test: 4000
  - Production: 8080
- Update `serve` script to serve `/docs` in production mode
- Add `serve:dev` script to serve `/docs` in development mode
- Fix development rendering issue caused by `DocsifyCarbon.create` error.
- Add server port checking to prevent potential test/CI issues from BrowserSync's port auto port selection behavior
- Update test documentation in `/test/README.md`
- Remove `index.html` file in root (was confusing, now no longer needed)
- Remove live-server dependency



## Related issue, if any:

#2218

## What kind of change does this PR introduce?

Other

## For any code change,



- [x] Related documentation has been updated, if needed
- [x] Related tests have been added or updated, if needed

## Does this PR introduce a breaking change?

No
